java/parser-hive/src/main/java/org/sqlflow/parser/hive/HiveParserAdaptor.java

```diff
@@ -2,15 +2,32 @@
 
 import java.lang.reflect.Field;
 import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
 import org.antlr.runtime.RecognitionException;
-import org.apache.hadoop.hive.ql.Context;
+import org.antlr.runtime.Token;
 import org.apache.hadoop.hive.ql.parse.ParseDriver;
+import org.apache.hadoop.hive.ql.parse.ParseDriver.ANTLRNoCaseStringStream;
+import org.apache.hadoop.hive.ql.parse.ParseDriver.HiveLexerX;
 import org.apache.hadoop.hive.ql.parse.ParseError;
 import org.apache.hadoop.hive.ql.parse.ParseException;
-import org.sqlflow.parser.parse.ParseInterface;
-import org.sqlflow.parser.parse.ParseResult;
-
-public class HiveParserAdaptor implements ParseInterface {
+import org.sqlflow.parser.parse.BaseParser;
+
+public class HiveParserAdaptor extends BaseParser {
+  private static Field errorsField;
+  private static Field reField;
+  private ParseDriver parseDriver = new ParseDriver();
+
+  static {
```

Comment:
"Static block" is java specific grammar, maybe change this into a constructor so that it's easier to read.

Reply:
we may keep this, because we want reField be static for better performance, and static block is recommend way to init static field

Reply:
Maybe use some static member and initialize it in the constructor if it's not initialized.